analytics/analytics_package/analytics/sheets_api.py

```diff
@@ -0,0 +1,244 @@
+    """
+    try:
+        gc.open(sheet_name)
+        return True
+    except gspread.exceptions.SpreadsheetNotFound:
+        return False
+
+def execute_drive_list(drive_api, search_params):
+    """
+    Execute a files().list() request on the Drive API with the given search parameters.
+    Returns the 'files' components of the response.
+
+    Positional arguments:
+    :param drive_api: the Drive API object
+    :param search_params: the search parameters, see https://developers.google.com/drive/api/v3/search-files
+    :returns: the 'files' components of the response
+    """
+    files_found = []
+    page_token = None
+    while True:
+        request = drive_api.files().list(q=search_params, spaces="drive", pageToken=page_token)
+        response = request.execute()
+        page_token = response.get("nextPageToken", None)
+        files_found += response.get("files", [])
+        if page_token is None:
+            break
+    return files_found
+
+def search_for_folder_id(drive_api, folder_name, allow_trashed = False, allow_duplicates = False):
+    """
+    Search for a folder by name in the Drive API.
+    Returns a list of folder ids that match the search criteria.
+
+    :param drive_api: the Drive API object
+    :param folder_name: the name of the folder to search for
+    :param allow_trashed: whether to include trashed folders in the search, defaults to False
+    :param allow_duplicates: whether to allow multiple folders with the same name, defaults to False
+    :returns: a list of folder ids that match the search criteria
+    """
+    search_params = f"name = '{folder_name}' and mimeType = 'application/vnd.google-apps.folder'"
+    if not allow_trashed:
+        search_params += " and trashed = false"
+    
+    files = execute_drive_list(drive_api, search_params)
+    files_exact_match = tuple(filter(lambda file: file["name"] == folder_name, files))
+    
+    if len(files_exact_match) > 1:
+        if not allow_duplicates:
+            raise RuntimeError("Too many files returned")
+    
+    return [file["id"] for file in files_exact_match]   
+
+
+def create_sheet_in_folder(authentication_response, sheet_name, parent_folder_name=None, override_behavior=FILE_OVERRIDE_BEHAVIORS.EXIT_ANYWHERE):
+    """
+    Create a new sheet in the project with the given name and parent folder.
+    Returns the new sheet.
+
+    :param authentication_response: the service parameters tuple
+    :param sheet_name: the name of the new sheet
+    :param parent_folder_name: the name of the parent folder for the new sheet
+    :param override_behavior: the behavior to take if the sheet already exists
+    :returns: the gspread.Spreadsheet object of the new sheet
+    :rtype: gspread.Spreadsheet
+    """
+    # Build Drive API
+    drive_credentials = extract_credentials(authentication_response)
+    gc = gspread.authorize(drive_credentials)
+    drive_api = build('drive', 'v3', credentials=drive_credentials)
```

Comment:
I imagine that, since `authenticate_gspread` and `authenticate_drive_api` exist, they should be used here instead of initializing the API clients directly